gammapy/cube/psf_map.py

```diff
@@ -331,3 +333,51 @@ def stack(self, other):
     def copy(self):
         """Copy PSFMap"""
         return deepcopy(self)
+
+    def sample_coord(self, map_coord, random_state=0):
+        """ Apply PSF corrections on the coordinates of a set of simulated events.
+
+        Parameters
+        ----------
+        map_coord : `~gammapy.maps.MapCoord` object.
+            Sequence of coordinates and energies of sampled events.
+        random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}
+            Defines random number generator initialisation.
+            Passed to `~gammapy.utils.random.get_random_state`.
+
+        Returns
+        -------
+        corr_coord : `~gammapy.maps.MapCoord` object.
+            Sequence of PSF-corrected coordinates of the input map_coord map.
+        """
+
+        random_state = get_random_state(random_state)
+
+        coord = {
+            "lon": map_coord.lon.reshape(-1, 1),
+            "lat": map_coord.lat.reshape(-1, 1),
+            "energy": map_coord["energy"].reshape(-1, 1),
+            "theta": (self.psf_map.geom.axes[0].center),
+        }
+
+        pdf = self.psf_map.interp_by_coord(coord)
+
+        sample_pdf = InverseCDFSampler(pdf, axis=1)
+        pix_coord = sample_pdf.sample_axis()
+        offset = self.psf_map.geom.axes[0].pix_to_coord(pix_coord)
+
+        phi_angle = random_state.uniform(360, size=len(map_coord.lon)) * u.deg
+        event_positions = SkyCoord(
```

Comment:
You could just use `map_coord.skycoord` to get a `SkyCoord` object, this will get the frame handling right.